scripts/compile_ilab_qa.py

```diff
@@ -0,0 +1,79 @@
+#!/bin/env python3
+# just 'python' doesn't work in container environment -_^
+# We are using ilab 0.18 schema, version 3
+
+import json
+import textwrap
+from typing import Optional
+
+import yaml
+import glob
+import pathlib
+import sys
+
+try:
+    LOGDETECTIVE_DATA_DIR = pathlib.Path(sys.argv[1])
+except IndexError:
+    print(
+        f"usage: {sys.argv[0]} PATH\n\n"
+        "First argument (PATH) is a path to directory tree of unpacked"
+          " results from the https://logdetective.com/download website.")
+    sys.exit(1)
+
+data = {
+    "version": 3,
+    "created_by": "Log Detective Team",
+    "domain": "software",
+    # dropped in version 3
+    # "task_description": "Annotated snippets from software logs that explain problems",
+    "seed_examples": [],
+    "document_outline": "Building RPMs",
+    "document": {
+        "repo": "https://github.com/fedora-copr/logdetective-taxonomy",
+        "commit": "HEAD",
+        "patterns": ["README.md"]
+    }
+}
+
+raw = []
+# ilab doesn't allow duplicate entries, so we need to make our entries unique
+haz_snippets = set()
+for file in glob.glob(f"{LOGDETECTIVE_DATA_DIR}/**/*.json", recursive=True):
+    with open(file) as f:
+        raw.append(json.load(f))
+
+for e in raw:
```

Comment:
Given the many calls to the `wrap` function. I would recommend instantiating a TextWrapper object[3].
The performance gain may be marginal however.

[3] https://docs.python.org/3/library/textwrap.html#textwrap.TextWrapper